python-backend/app/api/applications/resources/apllication.py

```diff
@@ -0,0 +1,91 @@
+from flask_restplus import Resource, reqparse
+from flask import request
+from datetime import datetime
+
+from app.extensions import api
+from ..models.application import Application
+from ...mines.mine.models.mine import Mine
+from app.api.utils.resources_mixins import UserMixin, ErrorMixin
+from app.api.utils.access_decorators import requires_role_mine_view, requires_role_mine_create
+
+
+class ApplicationResource(Resource, UserMixin, ErrorMixin):
+    parser = reqparse.RequestParser()
+    parser.add_argument('application_no', type=str, help='Number of the application being added.')
+    parser.add_argument('mine_guid', type=str, help='guid of the mine.')
+    parser.add_argument(
+        'application_status_code',
+        type=str,
+        help='Status of the application being added.',
+        store_missing=False)
+    parser.add_argument(
+        'received_date', type=lambda x: datetime.strptime(x, '%Y-%m-%d') if x else None)
+    parser.add_argument(
+        'description', type=str, help='Application description', store_missing=False)
+
+    @api.doc(params={'application_guid': 'Application guid.'})
+    @requires_role_mine_view
+    def get(self, application_guid=None):
+
+        mine_guid = request.args.get('mine_guid', None, type=str)
+        if mine_guid:
+            applications = Application.find_by_mine_guid(mine_guid)
+            result = [app.json() for app in applications]
+        elif application_guid:
```

Comment:
I've done this reversed, if someone provides a guid in the url, that should supersede any query string parameters.

Reply:
True, I'll swap that around.